src/main/java/uk/gov/justice/hmpps/casenotes/services/ExternalApiService.java

```diff
@@ -98,7 +99,7 @@ String getOffenderLocation(final String offenderIdentifier) {
     private int getHeader(final ResponseEntity response) {
         final var responseHeaders = response.getHeaders();
         final var value = responseHeaders.get("Total-Records");
-        return value != null && !value.isEmpty() ? Integer.valueOf(value.get(0)) : 0;
+        return value != null && !value.isEmpty() ? Integer.parseInt(value.get(0)) : 0;
```

Comment:
For future: something like this would be more readable:


`Optional.ofNullable(value).filter(not(HttpHeader::isEmpty)).map(HttpHeader::findFirst).orElse(0)`

Reply:
Agreed, I just accepted the intellij suggestion about the warning.
In this instance the `value` is a list of strings rather than a `HttpHeader` though, so
```java
Optional.ofNullable(value).flatMap(v -> v.stream().findFirst()).map(Integer::parseInt).orElse(0);
```
would work